ql/src/java/org/apache/hadoop/hive/ql/txn/compactor/CompactorFactory.java

```diff
@@ -99,10 +99,12 @@ public Compactor getCompactor(Table table, HiveConf configuration, CompactionInf
         case MINOR:
           return new MmMinorQueryCompactor();
         case MAJOR:
+        case REBALANCE:
+          // REBALANCE COMPACTION on an insert-only table is simply a MAJOR compaction. Since there is no ACID row data,
+          // there is no acid row order to keep, and the number of buckets cannot be set at all (it will be calculated
+          // and created by TEZ dynamically). Initiator won't schedule REBALANCE compactions for insert-only tables,
+          // however users can request it. In these cases we simply fall back to MAJOR compaction
           return new MmMajorQueryCompactor();
-        default:
```

Comment:
why remove this, what if the user mistypes?

Reply:
In that case the execution will jump to the last line of the method and throw an exception. However now I realized that the message needs to be actualized.